src/lib.rs

```diff
@@ -74,41 +86,73 @@ use errors::*;
 
 mod diff;
 
-/// Assertions for a specific command
+/// Assertions for a specific command.
 #[derive(Debug)]
 pub struct Assert {
     cmd: Vec<String>,
-    expect_success: bool,
+    expect_success: Option<bool>,
```

Comment:
I agree that `succeeds` _is_ a reasonable default and symmetry is not always the best design. But
- you would then need an un-setter `.fails_or_succeeds()` to express the I-don't-care-case
- it should be well-documented

People may have very different expectations about this. So either way there might be confusion about the behaviour if it's not well-documented.

That could also be a candidate for a poll in users.rust-lang.org. ;-)